backend/czi_hosted/data_common/matrix_loader.py

```diff
@@ -152,6 +155,21 @@ def data_adaptor(self, url_dataroot, location, app_config):
                 cache_item = info.cache_item
 
             if data_adaptor is None:
+                if app_config.server_config.data_locator__api_base:
+                    if url_dataroot == "e":  # only pull dataset identification information from the portal for /e/ route
+                        headers = {"Content-Type": "application/json"}
+                        curr_url = f"{app_config.server_config.get_web_base_url()}/{url_dataroot}/{dataset}"
```

Comment:
`dataset` can be None, in which case I suspect the request below will not behave correctly.  Missing check?